utils/appliance/__init__.py

```diff
@@ -348,18 +355,25 @@ def configure(self, log_callback=None, **kwargs):
         """
 
         log_callback("Configuring appliance {}".format(self.address))
+        loosen_pgssl = kwargs.pop('loosen_pgssl', True)
+        fix_ntp_clock = kwargs.pop('fix_ntp_clock', True)
+        region = kwargs.pop('region', 0)
+        key_address = kwargs.pop('key_address', None)
         with self as ipapp:
             ipapp.wait_for_ssh()
-            self._configure_upstream(log_callback=log_callback)
-
-    def _configure_upstream(self, log_callback=None):
-        self.wait_for_evm_service(timeout=1200, log_callback=log_callback)
-        self.deploy_merkyl(start=True, log_callback=log_callback)
-        self.fix_ntp_clock(log_callback=log_callback)
-        self.setup_upstream_db(log_callback=log_callback)
-        self.loosen_pgssl(log_callback=log_callback)
-        self.restart_evm_service(log_callback=log_callback)
-        self.wait_for_web_ui(timeout=1800, log_callback=log_callback)
+            self.deploy_merkyl(start=True, log_callback=log_callback)
+            if fix_ntp_clock:
+                self.fix_ntp_clock(log_callback=log_callback)
+                # TODO: Handle external DB setup. Also in setup_database
+            self.setup_database(
+                region=region,
+                key_address=key_address,
+                log_callback=log_callback)
+            self.wait_for_evm_service(timeout=1200, log_callback=log_callback)
+            if loosen_pgssl:
+                self.loosen_pgssl(log_callback=log_callback)
+                self.restart_evm_service(log_callback=log_callback)
```

Comment:
So, when you configure the DB, the watchdog will fire up the evmserverd automatically, so then you don't need to do it. When you mess with the DB it is better to restart it. That is why I moved it into the if.

We aren't going to disable the loosening anyway soon.